src/raftstore/store/peer_storage.rs

```diff
@@ -762,14 +763,16 @@ impl PeerStorage {
         };
 
         for entry in entries {
-            if entry.get_sync_log() {
-                ready_ctx.sync_log = true;
-            } else {
-                let ctx = entry.get_context();
-                if !ctx.is_empty() {
-                    let ctx = ProposalContext::from_bits_truncate(ctx[0]);
-                    if ctx.contains(ProposalContext::SYNC_LOG) {
-                        ready_ctx.sync_log = true;
+            if !ready_ctx.sync_log {
```

Comment:
reduce the indentation
```
if ready_ctx.sync_log {
    continue
}
```

Reply:
We needs to save msgs. https://github.com/pingcap/tikv/pull/3045/files/4f254b50b100ead30e8471207cbf23d7a3bc0dfb#diff-ac5fa05712c478a85585b4852858d036R779

Reply:
I can't get the point, here you only need to update ready context sync log. any other msgs?

Reply:
https://github.com/pingcap/tikv/blob/1ef68e137e2776c1b937c41148de56b4b3332b84/src/raftstore/store/peer_storage.rs#L765-L783

Reply:
```
for entry in entries { 
     if ready_ctx.sync_log {
         continue
     } 
         if entry.get_sync_log() { 
             ready_ctx.sync_log = true; 
         } else { 
             let ctx = entry.get_context(); 
             if !ctx.is_empty() { 
                 let ctx = ProposalContext::from_bytes(ctx); 
                 if ctx.contains(ProposalContext::SYNC_LOG) { 
                     ready_ctx.sync_log = true; 
                 } 
             } 
     } 
```

Reply:
Then we will miss:

```rust
            ready_ctx.raft_wb.put_msg(
                &keys::raft_log_key(self.get_region_id(), entry.get_index()),
                entry,
            )?;
```

Reply:
Why not add a function called `get_sync_log_from_entry`?